✨ [RUM-3710] Update session ID handling to support cookie deletion (#2673)
DataDog/browser-sdk: packages/core/src/domain/session/README.md

Author: tip: use _Display the rich diff_ on this file

## possible states:

| state      | value                                                        |
| ---------- | ------------------------------------------------------------ |
| NotStarted | `{}`                                                         |
| Expired    | `{expired: '1'}` or `expire >  15 minutes` or `created > 4H` |
| Tracked    | `{id: 'xxxx-xx-xx}`                                          |
| NotTracked | `{rum: 0}` or `{logs: 0}`                                    |

> ## session initializion

Author: ```suggestion
## session initialization
```

  fork_state --> NotStarted
  fork_state --> Expired
  fork_state --> Tracked
  fork_state --> NotTracked

  NotStarted --> [*]

  Expired --> fork_state2: renew()
  Tracked --> fork_state2: extend()
  NotTracked --> fork_state2: extend()

  fork_state2 --> fork_state3: computeTrackingType()

  fork_state3 --> Tracked2
  fork_state3 --> NotTracked2

  Tracked2 --> [*]
  NotTracked2 --> [*]
}
```

> [!NOTE]  
> Because `computeTrackingType()` happens at every `expandOrRenew()`, it is in theory possible to switch from `Tracked` to `NotTracked` and vice versa within an active session. However, this is not expected to happen in practice at this time.

## Extend Or Expire

```mermaid
stateDiagram-v2
state fork_state <<fork>>

state extendOrExpire() {
  [*] --> fork_state : extendOrExpire()
  Tracked2: Tracked
  NotTracked2: NotTracked
  Expired2: Expired

  fork_state --> NotStarted
  fork_state --> Expired
  fork_state --> Tracked
  fork_state --> NotTracked

  Tracked --> Tracked2: extend()
  NotTracked --> NotTracked2: extend()
  Expired --> Expired2: expire()

  NotStarted --> [*]
  Tracked2 --> [*]
  NotTracked2 --> [*]
  Expired2 --> [*]
}
```

> [!NOTE]  

> > Because the `Expired` state could happens because a session timed out, `expire()` will normalize explicitely the state to `isExpired=1`

Reviewer: ```suggestion
> Because a session time out can result in an `Expired` state, `expire()` explicitly normalizes the state to `isExpired=1`
```
